text/0000-router-link-and-helpers.md

```diff
@@ -0,0 +1,124 @@
+- Start Date: 2018-06-15
+- RFC PR:
+- Ember Issue:
+
+# Add a Link component and routing helpers
+
+## Summary
+
+The purpose of this is to add a single component, `Link`, that wraps up some routing helpers for default usage of linking / routing.
+
+## Motivation
+
+Positional arguments are not planned for use with Angle Bracket invocations. For clarity, and with some inspiration from react-router, It may be beneficial for Ember to include a `Link` component, similar to the existing `link-to`, but with only named arguments. This will help with clarity of intent, and also allow for easier transitioning from other communities who already have named-argument links.
+
+## Detailed design
+
+*Phase 1*
+- add built-in helpers for custom link-like behavior
+- add `Link` component with `@to` named argument -- all other named arguments will be the params
+
+*Phase 2*
+- deprecate `LinkTo`
+
+*Phase 3*
+- remove `LinkTo` and the corresponding deprecation
+
+### Details
+
+*Helpers to be implemented*
+
+ - `transition`
+    this is a helper that would tie into the router to be able to achieve the same functionality as the navigational usage of `link-to`
+
+    examples:
+
+      ```hbs
+      <button {{action (transition to='posts.edit' postId=post.id)}} />
```

Comment:
This is an accessibility red flag to me, links should be links. While the `transition` helper probably has value in other places I'd like to see an example that doesn't just turn a button into a styled link.